packages/twenty-front/src/modules/object-record/record-field/hooks/useClearField.ts

```diff
@@ -0,0 +1,90 @@
+import { useContext } from 'react';
+import { useRecoilCallback } from 'recoil';
+
+import { objectMetadataItemsState } from '@/object-metadata/states/objectMetadataItemsState';
+import { recordStoreFamilySelector } from '@/object-record/record-store/states/selectors/recordStoreFamilySelector';
+import { generateEmptyFieldValue } from '@/object-record/utils/generateEmptyFieldValue';
+import { generateEmptyFieldValueForClear } from '@/object-record/utils/generateEmptyFieldValueForClear';
+
+import { FieldContext } from '../contexts/FieldContext';
+import { isFieldRelation } from '../types/guards/isFieldRelation';
+
+export const useClearField = () => {
+  const {
+    entityId,
+    fieldDefinition,
+    useUpdateRecord = () => [],
+  } = useContext(FieldContext);
+
+  const [updateRecord] = useUpdateRecord();
+
+  const clearField = useRecoilCallback(
+    ({ snapshot, set }) =>
+      () => {
+        const fieldIsRelation = isFieldRelation(fieldDefinition);
+
+        if (fieldIsRelation) {
```

Comment:
Could you please extract the logic in a function such as `clearFieldRelation`? When a if logic is so big it makes it a bit hard to read